File: pset6/readability/readability.py
from cs50 import get_string

text = get_string("Text: ")

letters = 0
words = 1
sentences = 0

for i in range(len(text)):
    if text[i] >= 'A' and text[i] <= 'z':
        letters += 1
    elif text[i] == ' ':
        words += 1
    elif text[i] == '.' or text[i] == '?' or text[i] == '!':
        sentences += 1

    l = (letters / words) * 100
    s = (sentences / words) * 100

    index = 0.0588 * l - 0.296 * s - 15.8

if index < 1:
    print("Before Grade 1")
elif index > 16:
    print("Grade 16+")
else:
    print("Grade ", round(index))

File: pset2/readability/readability.c
#include <cs50.h>
#include <stdio.h>
#include <math.h>
#include <string.h>

int main(void)
{
    int i,x;
    int words = 1;
    int sentences = 0;
    int letters = 0;
    string text = get_string("Text: ");

    for(i = 0, x = strlen(text); i<x; i++)
    {
        if(text[i]>='A' && text[i]<='z')
        {
            letters++;
        }
        else if (text[i]==' ')
        {
            words++;
        }
        else if (text[i]=='.'||text[i]=='?'||text[i]=='!')
        {
            sentences++;
        }
    }

    float l = (letters / (float) words) * 100;
    float s = (sentences / (float) words) * 100;

    float index = 0.0588 * l - 0.296 * s - 15.8;

    if(index<1)
    {
        printf("Before Grade 1\n");
    }
    else if(index>16)
    {
        printf("Grade 16+\n");
    }
    else
    {
    printf("Grade %i\n",(int)round(index));
    }
}